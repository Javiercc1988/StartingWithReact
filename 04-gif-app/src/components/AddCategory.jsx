//Creacion de componente rapido rafc

import React, { useState } from "react";

export const AddCategory = ({ onAddNewCategory }) => {
  const [inputValue, setinputValue] = useState("");

  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const onSubmit = (event) => {
    // Evitar comportamiento de refresco de página por defecto del formulario
    event.preventDefault();

    const cleanValue = inputValue.trim();
    if (cleanValue.length <= 1) return;

    // Usamos la función enviada por props del componente para emitir al componente padre el valor
    // De esta forma se encarga el componente padre de trabajar con el dato
    onAddNewCategory(cleanValue);

    // Reseteamos el input
    setinputValue("");
  };

  return (
    // <form onSubmit={(event) => onSubmit(event)}>
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Introduce una categoría"
        value={inputValue}
        onChange={onInputChange}
        // onChange={(inputEvent) => onInputChange(inputEvent)}
      />
    </form>
  );
};
