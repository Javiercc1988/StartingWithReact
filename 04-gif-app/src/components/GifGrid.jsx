export const GifGrid = ({ category }) => {
  const listado = [1, 2, 3, 4];
  return (
    <>
      <h3>{category}</h3>
      {
      
        listado.map((gif) => <p key={gif}>{gif}</p>)
        
    }
    </>
  );
};
