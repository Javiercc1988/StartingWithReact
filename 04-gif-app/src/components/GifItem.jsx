// Los prop-types con vite tenemos que instalarlos: npm i prop-types
import PropTypes from "prop-types";

export const GifItem = ({ id, title, url }) => {
  return (
    <article className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </article>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
