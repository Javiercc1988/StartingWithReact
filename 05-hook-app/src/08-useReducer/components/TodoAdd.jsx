import { useForm } from "../../hooks";

export const TodoAdd = ({ handleNewTodo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: "",
  });

  const onNewTodo = (event) => {
    event.preventDefault();

    const cleanValue = description.trim();
    if (cleanValue.length <= 1) return;

    const newTodo = {
      id: new Date().getTime(),
      todo: description,
      done: false,
    };

    handleNewTodo(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={onNewTodo}>
      <input
        type="text"
        placeholder="¿Qué hay que hacer?"
        className="form-control"
        name="description"
        value={description}
        onChange={onInputChange}
      />

      <button type="submit" className="btn btn-primary mt-1">
        Agregar
      </button>
    </form>
  );
};
