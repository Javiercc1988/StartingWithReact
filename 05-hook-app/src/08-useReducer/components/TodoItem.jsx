export const TodoItem = ({ onDeleteTodo, todo, onToggleTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between">
      <span
        className={`align-self-center ${
          todo.done ? "text-decoration-line-through" : ""
        }`}
        onClick={() => onToggleTodo(todo)}
      >
        {todo.todo}
      </span>
      <button onClick={() => onDeleteTodo(todo)} className="btn btn-danger">
        Borrar
      </button>
    </li>
  );
};
