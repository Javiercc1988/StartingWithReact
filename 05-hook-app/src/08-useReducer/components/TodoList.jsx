import { TodoItem } from "./TodoItem";

export const TodoList = ({ onDeleteTodo, onToggleTodo, todos = [] }) => {

  return (
    <ul className="list-group">
      {todos.map((todo) => {
        return (
          <TodoItem
            key={todo.id}
            todo={todo}
            onDeleteTodo={onDeleteTodo}
            onToggleTodo={onToggleTodo}
          />
        );
      })}
    </ul>
  );
};
