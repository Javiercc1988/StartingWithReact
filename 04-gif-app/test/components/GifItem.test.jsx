import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba componente GifItem", () => {
  const url = "http://localhost/url_de_pruebas";
  const title = "Titulo de pruebas";

  test("Debe hacer match con el snapshot ", () => {
    const { container } = render(<GifItem url={url} title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem url={url} title={title} />);
    // screen.debug() Nos permite imprimir por pantalla el cÃ³digo del componente

    // Forma larga de hacerlo
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("El article debe tener la clase 'card' ", () => {
    render(<GifItem url={url} title={title} />);
    expect(screen.getByRole("article").className).toBe("card");
  });
});
