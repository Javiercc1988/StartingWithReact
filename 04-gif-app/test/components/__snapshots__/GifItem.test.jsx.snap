// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente GifItem Debe hacer match con el snapshot  1`] = `
<div>
  <article
    class="card"
  >
    <img
      alt="Titulo de pruebas"
      src="http://localhost/url_de_pruebas"
    />
    <p>
      Titulo de pruebas
    </p>
  </article>
</div>
`;
